src/frontend/Semantic_error.ml

```diff
@@ -55,6 +56,12 @@ module TypeError = struct
     | IntOrRealExpected (name, ut) ->
         Fmt.pf ppf "%s must be of type int or real. Instead found type %a."
           name UnsizedType.pp ut
+    | TypeExpected (name, (UInt | UReal), ut) ->
+        Fmt.pf ppf "%s must be of type int or real. Instead found type %a."
```

Comment:
must be a scalar to make it shorter and consistent with the others?

Reply:
Agreed , "scalar" is better.